`cortex_m_rt::exception` is no longer hygenic in 0.7

Previously, one could apply `#[cortex_m_rt::exception]` fully qualified, as shown [here](https://github.com/cbiffle/lilos/blob/e4de81717b36a6fe4151f2db15a217829e9b998e/os/src/time.rs#L102).

With the release of 0.7.1, this has stopped working. For the macro to work, one must now `use cortex_m_rt::exception` and then use `#[exception]`. The error message is not great, but suggests that it's a problem with the macro referring to itself by an unqualified name internally:

```
error[E0433]: failed to resolve: use of undeclared crate or module `exception`
   --> src/time.rs:102:1
    |
102 | #[cortex_m_rt::exception]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `exception`
    |
    = note: this error originates in the attribute macro `cortex_m_rt::exception` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0433`.
```

(That message sent me on a bit of a goose chase figuring out why `exception` was not defined in `cortex_m_rt`, until I worked it out.)

--- a/cortex-m-rt/CHANGELOG.md
+++ b/cortex-m-rt/CHANGELOG.md
@@ -7,6 +7,17 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+## Fixes
+
+- Fix `cortex_m_rt::exception` macro no longer being usable fully-qualified ([#414])
+
+[#414]: https://github.com/rust-embedded/cortex-m/issues/414
+
+## Notes
+
+- From this release onwards, cortex-m-rt lives in the cortex-m repository;
+  issue numbers refer there.
+
 ## [v0.7.1]
 
 ## Fixes

--- a/cortex-m-rt/Cargo.toml
+++ b/cortex-m-rt/Cargo.toml
@@ -15,6 +15,7 @@ repository = "https://github.com/rust-embedded/cortex-m"
 version = "0.7.1"
 autoexamples = true
 links = "cortex-m-rt" # Prevent multiple versions of cortex-m-rt being linked
+edition = "2018"
 
 [dependencies]
 cortex-m-rt-macros = { path = "macros", version = "=0.7.0" }

--- a/cortex-m-rt/macros/src/lib.rs
+++ b/cortex-m-rt/macros/src/lib.rs
@@ -328,7 +328,7 @@ pub fn exception(args: TokenStream, input: TokenStream) -> TokenStream {
             f.block.stmts = iter::once(
                 syn::parse2(quote! {{
                     // check that this exception actually exists
-                    exception::#ident;
+                    cortex_m_rt::exception::#ident;
                 }})
                 .unwrap(),
             )

--- a/cortex-m-rt/macros/src/lib.rs
+++ b/cortex-m-rt/macros/src/lib.rs
@@ -174,7 +174,7 @@ pub fn exception(args: TokenStream, input: TokenStream) -> TokenStream {
         Exception::Other => {
             quote! {
                 const _: () = {
-                    let _ = cortex_m_rt::Exception::#ident;
+                    let _ = ::cortex_m_rt::Exception::#ident;
                 };
             }
         }
@@ -328,7 +328,7 @@ pub fn exception(args: TokenStream, input: TokenStream) -> TokenStream {
             f.block.stmts = iter::once(
                 syn::parse2(quote! {{
                     // check that this exception actually exists
-                    cortex_m_rt::exception::#ident;
+                    ::cortex_m_rt::exception::#ident;
                 }})
                 .unwrap(),
             )

